components/device-mgt/org.wso2.carbon.device.mgt.core/src/test/java/org/wso2/carbon/device/mgt/core/search/ProcessorImplTest.java

```diff
@@ -0,0 +1,174 @@
+import org.wso2.carbon.device.mgt.core.search.mgt.InvalidOperatorException;
+import org.wso2.carbon.device.mgt.core.search.mgt.SearchMgtException;
+import org.wso2.carbon.device.mgt.core.search.mgt.impl.ProcessorImpl;
+import org.wso2.carbon.device.mgt.core.search.util.ChangeEnumValues;
+import org.wso2.carbon.device.mgt.core.search.util.Utils;
+import org.wso2.carbon.device.mgt.core.service.DeviceManagementProviderService;
+import org.wso2.carbon.device.mgt.core.service.DeviceManagementProviderServiceImpl;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+
+import javax.sql.DataSource;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * This class holds unit test cases for org.wso2.carbon.device.mgt.core.search.mgt.impl.ProcessorImpl
+ * */
+public class ProcessorImplTest extends BaseDeviceManagementTest{
+
+    private DeviceAccessAuthorizationService deviceAccessAuthorizationService;
+    private static final Log log = LogFactory.getLog(SearchManagementServiceTest.class);
+    private static List<DeviceIdentifier> deviceIdentifiers = new ArrayList<>();
+    private static final String DEVICE_ID_PREFIX = "SEARCH-DEVICE-ID-";
+    private static final String DEVICE_TYPE = "SEARCH_TYPE";
+
+    @BeforeClass
+    public void init() throws Exception {
+        deviceAccessAuthorizationService = DeviceManagementDataHolder.getInstance()
+                .getDeviceAccessAuthorizationService();
+
+        for (int i = 0; i < 5; i++) {
+            deviceIdentifiers.add(new DeviceIdentifier(DEVICE_ID_PREFIX + i, DEVICE_TYPE));
+        }
+        DeviceManagementProviderService deviceMgtService = new DeviceManagementProviderServiceImpl();
+        DeviceManagementServiceComponent.notifyStartupListeners();
+        DeviceManagementDataHolder.getInstance().setDeviceManagementProvider(deviceMgtService);
+        deviceMgtService.registerDeviceType(new TestDeviceManagementService(DEVICE_TYPE,
+                MultitenantConstants.SUPER_TENANT_DOMAIN_NAME));
+
+        List<Device> devices = TestDataHolder.generateDummyDeviceData(deviceIdentifiers);
+        for (Device device : devices) {
+            device.setDeviceInfo(Utils.getDeviceInfo());
+            deviceMgtService.enrollDevice(device);
+        }
+        List<Device> returnedDevices = deviceMgtService.getAllDevices(DEVICE_TYPE, true);
+        for (Device device : returnedDevices) {
+            if (!device.getDeviceIdentifier().startsWith(DEVICE_ID_PREFIX)) {
+                throw new Exception("Incorrect device with ID - " + device.getDeviceIdentifier() + " returned!");
+            }
+        }
+    }
+
+    @Test(description = "Test the Search Processor")
+    public void testWithNoDeviceAccessAuthorization() throws Exception {
```

Comment:
Why are we throwing generic exception ? Throw only the specific exceptions.

Reply:
This is fixed in https://github.com/wso2/carbon-device-mgt/pull/1082/commits/b1b4b26dbe0d3bdf4bd5b169dcb63be827495517